ledger/src/blockstore_processor.rs

```diff
@@ -1195,30 +1195,38 @@ fn confirm_slot_entries(
         true, // shuffle transactions.
         transaction_status_sender,
         replay_vote_sender,
-        timing,
+        batch_execute_timing,
         log_messages_bytes_limit,
         prioritization_fee_cache,
     )
     .map_err(BlockstoreProcessorError::from);
-    replay_elapsed.stop();
-    timing.replay_elapsed += replay_elapsed.as_us();
-
-    // If running signature verification on the GPU, wait for that computation to finish, and get
-    // the result of it. If we did the signature verification on the CPU, this just returns the
-    // already-computed result produced in start_verify_transactions.  Either way, check the result
-    // of the signature verification.
-    if !check_result.finish_verify() {
-        warn!("Ledger proof of history failed at slot: {}", bank.slot());
-        return Err(TransactionError::SignatureFailure.into());
+    replay_timer.stop();
+    *replay_elapsed += replay_timer.as_us();
+
+    {
+        // If running signature verification on the GPU, wait for that computation to finish, and
+        // get the result of it. If we did the signature verification on the CPU, this just returns
+        // the already-computed result produced in start_verify_transactions.  Either way, check the
+        // result of the signature verification.
+        let valid = transaction_verification_result.finish_verify();
+
+        // The GPU Entry verification (if any) is kicked off right when the CPU-side Entry
+        // verification finishes, so these times should be disjoint
+        *transaction_verify_elapsed +=
+            transaction_cpu_duration_us + transaction_verification_result.gpu_verify_duration();
+
+        if !valid {
+            warn!(
+                "Ledger transaction signature verification failed at slot: {}",
+                bank.slot()
+            );
+            return Err(TransactionError::SignatureFailure.into());
+        }
```

Comment:
👍 I think this is a good change to accumulate these timings even if we run into a failure, b/c we still waited in `finish_verify`

Reply:
I was not sure how to best deal with them.
It is an edge case for the slot verification to actually fail, I assume, right?
Currently, if the signature verification fails, we are not updating the PoH verification counters (next paragraph in the code).

Both PoH and signature verification may run in parallel to the replay code.  So the time counters they would be non-zero even if we did not wait in the `finish_verify()` call.

I have [another PR](https://github.com/solana-labs/solana/pull/30807) to track "confirmation time" - wall clock for this whole method end to end.
And it is updated regardless, both for successful and failed executions.

Also, I have an entire branch, where I've moved most of the verification code into a separate service.  In which case, metrics for the verification process also move into this service and are more regular.

So, I figured, I could do a small improvement here.  But there is probably no reason to fix it all the way.

Reply:
that makes sense to me, smaller PR's are better